app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -326,14 +326,28 @@ export class CloneRepository extends React.Component<
     this.updateAndValidatePath(newPath)
   }
 
-  private async doesPathExist(path: string) {
-    const exists = await pathExists(path)
-    // If the path changed while we were checking, we don't care anymore.
-    if (this.state.path !== path) {
-      return
-    }
+  private async validateEmptyFolder(path: string): Promise<null | Error> {
+    try {
+      const directoryFiles = await readdir(path)
+
+      // If the path has changed we no longer want the result
+      if (path !== this.state.path) {
+        return new Error('The path has been changed during check.')
```

Comment:
For other cases like this, where the path has changed under us, I think we just return `null` to indicate no action, rather than returning an error. Does that make sense?

Reply:
I am currently thinking through the results of this change.

After making this change, in the situation of the path having changed while `validateEmptyFolder` runs, then:
- `validateEmptyFolder` would return null
- The caller of `validateEmptyFolder` would then set the UI `Error` state null

If I understand correctly, then the user would not see any error and not be blocked from attempting to cloning to a path, even though the path has not yet been validated.
___
I am not sure if this is the expected behavior.
I would like to make sure we are on the same page on this change before making a commit for this.

However, I may be thinking through this incorrectly.

Reply:
I think we should check this before updating the state, but that clashes with what we were talking about on the previous line

Reply:
I think the best scenario we have would be to display an error when the path has changed and block cloning, perhaps telling them to try again.

If we don't display an error of some sort when the path has not yet been validated then they will be able to clone (potentially failing because the path can be anything). If the `Error` state does not get updated in this situation, it might potentially remain as no error and allow cloning.

Previous to my changes if a path has been changed during the check it would simply display the generic "The destination already exists." Perhaps "Failed to validate the entered path. Please try again."

Reply:
> I think the best scenario we have would be to display an error when the path has changed and block cloning, perhaps telling them to try again.

I still don't think this is the right option - as a user it doesn't matter to me if the path has changed while doing this check, I'd just like the app to check the path I chose.

> If we don't display an error of some sort when the path has not yet been validated then they will be able to clone (potentially failing because the path can be anything).

Ideally there would be an updated check coming through where `path === this.state.path` - if this isn't the case then maybe there's a race condition in here?

Reply:
ok so here's my take: the problem is that we use `this.state.path` to control what is displayed in the text field the user is typing in. therefore, i think we should never override that. the error state is separate from that and i think its ok if its not guaranteed to be immediately correct (it should only be delayed a little bit).

i think this check (`path !== this.state.path`) is unnecessary (and confusing) for this method. we should simply perform the validation here and let the state concerns be handled by callers. (i'm pretty sure i've indicated something contrary to this in the past so sorry for missing some of this nuance upfront!)